pkg/workflows/wasm/workflowLib/plugin.go

```diff
@@ -0,0 +1,270 @@
+			return nil, err
+		}
+
+		prevOutcome := &pb.Outcome{}
+		if err := proto.Unmarshal(outctx.PreviousOutcome, prevOutcome); err != nil {
+			return nil, err
+		}
+
+		for id, currSeqNum := range observation.Requests {
+			if err := ctx.Err(); err != nil {
+				return nil, err
+			}
+			if _, ok := prevOutcome.ObservedDonTimes[id]; !ok {
+				prevOutcome.ObservedDonTimes[id] = &pb.ObservedDonTimes{}
+			}
+			if currSeqNum == int64(len(prevOutcome.ObservedDonTimes[id].Timestamps)) {
+				NumFinishedRequests[id]++
+			}
+		}
+
+		for _, id := range observation.Finished {
+			finishedNodes[id]++
+		}
+
+		times = append(times, observation.Timestamp)
+	}
+
+	p.lggr.Debugw("Observed Node Timestamps", "timestamps", times)
+	slices.Sort(times)
+	donTime := times[len(times)/2]
+
+	if err := ctx.Err(); err != nil {
+		return nil, err
+	}
+
+	outcome := &pb.Outcome{}
+	if err := proto.Unmarshal(outctx.PreviousOutcome, outcome); err != nil {
+		return nil, err
+	}
+
+	if outcome.FinishedExecutionRemovalTimes == nil {
+		outcome.FinishedExecutionRemovalTimes = make(map[string]int64)
+	}
+	if outcome.ObservedDonTimes == nil {
+		outcome.ObservedDonTimes = make(map[string]*pb.ObservedDonTimes)
+	}
+	if outcome.RemovedExecutionIDs == nil {
+		outcome.RemovedExecutionIDs = make(map[string]bool)
+	}
+
+	// Compare with prior outcome to ensure DON time never goes backward.
+	if donTime < outcome.Timestamp+p.minTimeIncrease {
+		p.lggr.Infow("DON Time incremented by minimum time increase to ensure time progression", "minTimeIncrease", p.minTimeIncrease)
+		donTime = outcome.Timestamp + p.minTimeIncrease
+	}
+
+	p.lggr.Infow("New DON Time", "donTime", donTime)
+	outcome.Timestamp = donTime
+
+	for id, numRequests := range NumFinishedRequests {
+		fmt.Println("NumRequests: ", id, numRequests)
+		p.lggr.Debugw("Checking finished requests", "executionID", id, "numRequests", numRequests)
+		if numRequests > int64(p.config.F) {
+			observedDonTimes, ok := outcome.ObservedDonTimes[id]
+			if !ok {
+				observedDonTimes = &pb.ObservedDonTimes{}
+			}
+			observedDonTimes.Timestamps = append(observedDonTimes.Timestamps, donTime)
+			outcome.ObservedDonTimes[id] = observedDonTimes
+		}
+	}
+
+	// Check if consensus is reached on the workflow execution being finished
+	for id, numFinished := range finishedNodes {
+		if numFinished >= int64(p.config.F) {
+			if _, ok := outcome.FinishedExecutionRemovalTimes[id]; ok {
+				continue
+			}
+			outcome.FinishedExecutionRemovalTimes[id] = donTime + p.offChainConfig.ExecutionRemovalTime.AsDuration().Milliseconds()
+		}
+	}
+
+	for id, removeAt := range outcome.FinishedExecutionRemovalTimes {
+		if removeAt <= donTime {
+			outcome.RemovedExecutionIDs[id] = true
+			delete(outcome.FinishedExecutionRemovalTimes, id)
+		}
+	}
+
+	return proto.Marshal(outcome)
+}
+
+func (p *workflowLibPlugin) Reports(ctx context.Context, seqNr uint64, outcome ocr3types.Outcome) ([]ocr3types.ReportPlus[struct{}], error) {
+	return nil, nil
```

Comment:
If you return nil here then nothing is going to be passed to the Transmitter. You need to pass unchanged outcome.